mycpp/gc_alloc.h

```diff
@@ -26,6 +26,11 @@ class StackRoots {
  public:
   // Note: void** seems logical, because these are pointers to pointers, but
   // the C++ compiler doesn't like it.
+  explicit StackRoots(void* root) {
+    n_ = 1;
```

Comment:
does removing `n_` altogether do anything?

Then the destructor can have no loop

Maybe it will be optimized better because the class  no longer contains any members at all?